### DISCUSSION TITLE
Errors+Crash when loading extension

### DISCUSSION THREAD

**Discussion by ImGGAAVVIINN** [tsukasa/BdBrowser#37] [Problems and issues]
![image](https://github.com/tsukasa/BdBrowser/assets/68761233/5292b23f-6ee5-4fdd-950d-5e3a2ff075b9)
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'get')
Context
https://discord.com/channels/982185912903561216/1072780738577059840
Stack Trace
js/frontend.js:4639 (anonymous function)  
  
   
![image](https://github.com/tsukasa/BdBrowser/assets/68761233/77e93c1a-8c92-4077-9b71-0acc2b00b2b5)
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'get')
Context
https://discord.com/channels/1015060230222131221/1024351821873037462
Stack Trace
js/frontend.js:4639 (anonymous function)  
  
![image](https://github.com/tsukasa/BdBrowser/assets/68761233/a81513e3-2cc8-40f1-8ec2-213fdfae5942)
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'get')
Context
https://discord.com/app
Stack Trace
js/frontend.js:4639 (anonymous function)  
  
![image](https://github.com/tsukasa/BdBrowser/assets/68761233/438e7f65-26be-4892-8e7e-5e07a3a6ff76)
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'get')
Context
https://discord.com/channels/1015060230222131221/1222936386626129920/1252339792390393856
Stack Trace
js/frontend.js:4639 (anonymous function)  
  
Browser: Vivaldi 6.6.3271.61 (Stable channel) stable (64-bit). But the error also occurs in, Version 126.0.6478.114 (Official Build) (64-bit), the latest Chrome version at the time of writing
OS: Ubuntu 20.04

**Comment by tsukasa** [comment]
Greetings,

thank you so much for the report! Discord has re-rolled its internals once again. Access to the StorageModule is no longer as easy as it was before.

Instead of trying to play ball with Discord, BdBrowser will now save an instance of the `window.localStorage` during preload, pick this instance up in the frontend and delete the reference so plugins cannot easily gain localStorage access.

I hope that this implementation proves a bit more durable in the long run.

The fixed version [1.10.2.20240619 is available from the releases](https://github.com/tsukasa/BdBrowser/releases/tag/v1.10.2.20240619).

**Comment by ImGGAAVVIINN** [comment]
Unfortunately, I am still getting errors and the extension won't load, I am not sure if I did something wrong
Errors:
![image](https://github.com/tsukasa/BdBrowser/assets/68761233/43d5757c-e444-4c9b-99c9-7c8f270b89b3)
%c[BDBrowser]%c %c[AsarUpdater]%c color: #3E82E5; font-weight: 700; color: #396CB8 Failed to download BetterDiscord Error: ENOENT: no such file or directory, open 'AppData/BetterDiscord/data/betterdiscord.asar'
**Context**
https://discord.com/channels/@me
**Stack Trace**
js/frontend.js:152 (#log)
js/frontend.js:168 (error)
js/frontend.js:690 (downloadBetterDiscordAsar)
![image](https://github.com/tsukasa/BdBrowser/assets/68761233/4aedd067-4111-423c-affe-47e518dec54c)
Uncaught (in promise) Error: BdBrowser Error: Downloading betterdiscord.asar or writing into VFS failed!
**Context**
https://discord.com/channels/@me
**Stack Trace**
js/frontend.js:6708 (anonymous function)
